Shopify/hydrogen
# Fix the Pagination component to use forwardRefs

…PreviousLink render props



### WHY are these changes introduced?

The load more pages on scroll example is broken because the `NextPage` and `PreviousPage` components are not using `forwardRef`s. 



### WHAT is this pull request doing?
Fix the Pagination component to use forwardRefs on the `NextPage` and `PreviousPage` components. Also fix the type definitions for each.




### HOW to test your changes?

Load a collection page on the demo store and make sure that pages load as you scroll down the page.

#### Post-merge steps



#### Checklist
